In [373]:
import numpy as np
import pandas as pd
import matplotlib as plt
%matplotlib inline
import seaborn as sns
import plotly.express as px
import dask.dataframe as dd
datadir= 'data/'
import matplotlib.pyplot as plt


In [374]:
infile = datadir + 'Disability_and_Health_Data_System__DHDS_.csv'
## realized the data was to large
infile = pd.read_csv(infile)

# Overview:

The year 2020 was a difficult year I want to analyze the CDC data to further understand how many people in our nation have disablities and what kind. 

I would like to explore the CDC Disability and Health Data Systems DHDS and create visualizations regading mental, physical, social econmical, racial demograpfic within that year. 

I had to change my questions from my proposal after going through the data to correlate it.



## Visualization Goal:
1. Disablity data 
2. Data regarding age
3. Data regarding gender
4. Racial disparities  


## Questions I wish to address

- What percentage of peoples over the age of 18 have a disablity?

- What percentage of adults have cognative disablity compared to visable?

- How does the data vary based on location?

- How does race correlate with individuals a disablity?

### DATA

"Disability and Health Data System (DHDS) is an online source of state-level data on adults with disabilities. Users can access information on six functional disability types: cognitive (serious difficulty concentrating, remembering or making decisions), hearing (serious difficulty hearing or deaf), mobility (serious difficulty walking or climbing stairs), vision (serious difficulty seeing), self-care (difficulty dressing or bathing) and independent living (difficulty doing errands alone)."

In [375]:
print('Nbr of records that have valid data since 2016: {0}'.format(len(infile)))

Nbr of records that have valid data since 2016: 438788


## First Data Cleaning:

I want to make my data frame focus on certain rows.

In [376]:
# settings to display all columns
pd.set_option("display.max_columns", None)
# display the dataframe head
infile.head()

Rowid  Year LocationAbbr LocationDesc  \
0   2016~31~JOB04~JOB~DISSTAT~NODIS~~~~  2016           NE     Nebraska   
1   2016~21~JOB04~JOB~DISSTAT~NODIS~~~~  2016           KY     Kentucky   
2  2016~66~JOB04~JOB~DISTYPE~VISDIS~~~~  2016           GU         Guam   
3  2016~41~JOB04~JOB~DISTYPE~VISDIS~~~~  2016           OR       Oregon   
4  2016~19~JOB04~JOB~DISTYPE~MOBDIS~~~~  2016           IA         Iowa   

  DataSource      Category                                          Indicator  \
0      BRFSS  Demographics  Employment status among adults 18 years of age...   
1      BRFSS  Demographics  Employment status among adults 18 years of age...   
2      BRFSS  Demographics  Employment status among adults 18 years of age...   
3      BRFSS  Demographics  Employment status among adults 18 years of age...   
4      BRFSS  Demographics  Employment status among adults 18 years of age...   

  Response Data_Value_Unit          Data_Value_Type  Data_Value  \
0    Other               %  Age-adjusted Prevalence        26.0   
1    Other               %  Age-adjusted Prevalence        30.5   
2    Other               %  Age-adjusted Prevalence        29.5   
3    Other               %  Age-adjusted Prevalence        27.9   
4    Other               %  Age-adjusted Prevalence        25.5   

   Data_Value_Alt Data_Value_Footnote_Symbol Data_Value_Footnote  \
0            26.0                        NaN                 NaN   
1            30.5                        NaN                 NaN   
2            29.5                        NaN                 NaN   
3            27.9                        NaN                 NaN   
4            25.5                        NaN                 NaN   

   Low_Confidence_Limit  High_Confidence_Limit  Number  WeightedNumber  \
0                  24.8                   27.2  3339.0        273340.0   
1                  28.8                   32.2  2080.0        628016.0   
2                  21.0                   39.6    33.0          2355.0   
3                  20.4                   36.8    69.0         36768.0   
4                  20.5                   31.3   604.0        120768.0   

  StratificationCategory1      Stratification1 StratificationCategory2  \
0       Disability Status        No Disability                     NaN   
1       Disability Status        No Disability                     NaN   
2         Disability Type    Vision Disability                     NaN   
3         Disability Type    Vision Disability                     NaN   
4         Disability Type  Mobility Disability                     NaN   

  Stratification2 CategoryID IndicatorID  Geolocation  LocationID ResponseID  \
0             NaN      DEMOG         JOB          NaN          31      JOB04   
1             NaN      DEMOG         JOB          NaN          21      JOB04   
2             NaN      DEMOG         JOB          NaN          66      JOB04   
3             NaN      DEMOG         JOB          NaN          41      JOB04   
4             NaN      DEMOG         JOB          NaN          19      JOB04   

  DataValueTypeID StratificationCategoryID1 StratificationID1  \
0      AGEADJPREV                   DISSTAT             NODIS   
1      AGEADJPREV                   DISSTAT             NODIS   
2      AGEADJPREV                   DISTYPE            VISDIS   
3      AGEADJPREV                   DISTYPE            VISDIS   
4      AGEADJPREV                   DISTYPE            MOBDIS   

  StratificationCategoryID2 StratificationID2  
0                       NaN               NaN  
1                       NaN               NaN  
2                       NaN               NaN  
3                       NaN               NaN  
4                       NaN               NaN

In [377]:
# locate a recod to see info
infile.loc[40]

Rowid                                      2016~36~BMI03~BMI~DISTYPE~COGDIS~~~~
Year                                                                       2016
LocationAbbr                                                                 NY
LocationDesc                                                           New York
DataSource                                                                BRFSS
Category                                               Health Risks & Behaviors
Indicator                     Body mass index category among adults 18 years...
Response                                                             Overweight
Data_Value_Unit                                                               %
Data_Value_Type                                         Age-adjusted Prevalence
Data_Value                                                                 31.5
Data_Value_Alt                                                             31.5
Data_Value_Footnote_Symbol              

- Here I will drop the columns I do not need 

In [378]:
to_drop = ['Rowid', 'DataSource', 'Data_Value_Unit','WeightedNumber', 'Geolocation', 'Category', 'Indicator', 'Data_Value_Type', 'Data_Value_Footnote_Symbol', 'Data_Value_Footnote', 'StratificationCategory1', 'StratificationCategoryID2', 'Stratification2', 'Stratification2', 'StratificationID2', 'StratificationCategory2' ]
infile.drop(to_drop, inplace = True, axis = 1)
infile.head()

Year LocationAbbr LocationDesc Response  Data_Value  Data_Value_Alt  \
0  2016           NE     Nebraska    Other        26.0            26.0   
1  2016           KY     Kentucky    Other        30.5            30.5   
2  2016           GU         Guam    Other        29.5            29.5   
3  2016           OR       Oregon    Other        27.9            27.9   
4  2016           IA         Iowa    Other        25.5            25.5   

   Low_Confidence_Limit  High_Confidence_Limit  Number      Stratification1  \
0                  24.8                   27.2  3339.0        No Disability   
1                  28.8                   32.2  2080.0        No Disability   
2                  21.0                   39.6    33.0    Vision Disability   
3                  20.4                   36.8    69.0    Vision Disability   
4                  20.5                   31.3   604.0  Mobility Disability   

  CategoryID IndicatorID  LocationID ResponseID DataValueTypeID  \
0      DEMOG         JOB          31      JOB04      AGEADJPREV   
1      DEMOG         JOB          21      JOB04      AGEADJPREV   
2      DEMOG         JOB          66      JOB04      AGEADJPREV   
3      DEMOG         JOB          41      JOB04      AGEADJPREV   
4      DEMOG         JOB          19      JOB04      AGEADJPREV   

  StratificationCategoryID1 StratificationID1  
0                   DISSTAT             NODIS  
1                   DISSTAT             NODIS  
2                   DISTYPE            VISDIS  
3                   DISTYPE            VISDIS  
4                   DISTYPE            MOBDIS

- Now I check to see if there are anymore to drop, there is they are: Category,Indicator,Data_Value_Type,Data_Value_Footnote_Symbol,Data_Value_Footnote, StratificationCategory1

- I will adjust to_drop above

- turn data frame infile into an array 

In [379]:
infile_array = infile.to_numpy()

## Q How many participants have a disability?

- then show disability type
- select specific columns from the data frame
- here I see that the years range from 2016-2019, I am thinking of spliting each column to to look like year & the type of disabilities 


In [383]:
#total occurance of disablities 
# infile Data Frame =  infile numpy array
# count total disablity 
countTD = (infile_array == 'DISTYPE').sum()

# the relationship between type disablities, each year


print("total number of disablities:",countTD)

total number of disablities: 109576


### A: the total number of partcipants with a disablity is 109576

## Q2 What kind of disablities were being reported? Which year had the most disablities?

In [391]:
infileDF = pd.DataFrame(infile)

In [393]:
# going to use dataframe.apply() to the all the rows that satisfy the condition 

infileDF = infileDF.apply(lambda x : True
                   if x['StratificationCategoryID1'] == 'DISTYPE' else False, axis = 1)

#count how many rows are true to the condition 
numRows = len(infileDF[infileDF == True].index)
print(numRows)
# this is also a better way of solving my first question and it validates the answer above

109576


The total rows that meet the condition are 109576 participants have a disablitiy over the years 2016-2019

Now I can try to see which year had the most disablities, dataFrame is now only a booling 
Because of the above condition so now will go back to infile and change it to a new data frame so I can categorize each year.

In [394]:
# dfyears is a df of infile 
dfyears = pd.DataFrame(infile)
#print(dfyears)
dfyears.groupby(['Year', 'Stratification1'], as_index=False).size()


Year                Stratification1   size
0   2016                 Any Disability  41470
1   2016           Cognitive Disability   6045
2   2016             Hearing Disability   6045
3   2016  Independent Living Disability   1950
4   2016            Mobility Disability   6045
5   2016                  No Disability  41470
6   2016                        Overall    520
7   2016           Self-care Disability   1950
8   2016              Vision Disability   6045
9   2017                 Any Disability  42240
10  2017           Cognitive Disability   5952
11  2017             Hearing Disability   5952
12  2017  Independent Living Disability   1920
13  2017            Mobility Disability   5952
14  2017                  No Disability  42240
15  2017                        Overall    512
16  2017           Self-care Disability   1920
17  2017              Vision Disability   5952
18  2018                 Any Disability  39552
19  2018           Cognitive Disability   5824
20  2018             Hearing Disability   5824
21  2018  Independent Living Disability   1920
22  2018            Mobility Disability   5824
23  2018                  No Disability  39552
24  2018                        Overall    512
25  2018           Self-care Disability   1920
26  2018              Vision Disability   5824
27  2019                 Any Disability  40320
28  2019           Cognitive Disability   5733
29  2019             Hearing Disability   5733
30  2019  Independent Living Disability   1890
31  2019            Mobility Disability   5733
32  2019                  No Disability  40320
33  2019                        Overall    504
34  2019           Self-care Disability   1890
35  2019              Vision Disability   5733

- Above I was able to group the years with the disablity to get the count for each
### My answer for the least amount of disablites is year 2018